dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImpl.java

```diff
@@ -1063,19 +1064,21 @@ protected Contentlet findContentletByIdentifierAnyLanguage(final String identifi
 
     @Override
     protected Contentlet findContentletByIdentifierAnyLanguage(final String identifier, final boolean includeDeleted) throws DotDataException, DotSecurityException {
-        // Looking content up this way can avoid any DB hits as these calls are all cached.
-        final List<Language> langs = this.languageAPI.getLanguages();
-        for(final Language language : langs) {
-            final Optional<ContentletVersionInfo> contentVersion = APILocator.getVersionableAPI().getContentletVersionInfo(identifier, language.getId());
-            if (contentVersion.isPresent() && UtilMethods.isSet(contentVersion.get().getIdentifier())
-                    && (includeDeleted || !contentVersion.get().isDeleted())) {
-                return find(contentVersion.get().getWorkingInode());
-            }
-            if (contentVersion.isPresent() && contentVersion.get().isDeleted() && !includeDeleted) {
-                Logger.warn(this, String.format("Contentlet with ID '%s' exists, but is marked as 'Archived'.",
-                        identifier));
-            }
+        final Optional<ContentletVersionInfo> contentVersionDeleted = FactoryLocator.getVersionableFactory()
```

Comment:
I would wrap this into a lazy to avoid unnecessary calls if the first if is true